Components/BasketHelper.php

```diff
@@ -223,9 +225,11 @@ public function fixBasket()
             $this->onlyConnectProducts = true;
 
             $this->removeNonProductsFromBasket();
-            $this->basket = $this->removeDefaultShipping($this->basket);
 
             $connectContent = $this->getConnectContent();
+            if (false === $this->customerHasToPayLocalShipping($connectContent)) {
```

Comment:
![image](https://user-images.githubusercontent.com/15930605/36036491-48dc99bc-0dba-11e8-85be-140327c13ad5.png)
Please don't use yoda conditions ;)

Reply:
In fact, this was a defined and wanted code style back when QaFoo was on the project, and in case of false or null there are good reasons to do it this way.

If you write false at the beginning of the statement, it is easy to scan the code for such conditions or to locate wrong logic. It is also the preferred way instead of writing "if(!$comething)" because the "!" is easy to miss.

So, use this for "false" and "null", but not for other constants.